kanbananza-backend/src/main/middleware/id_param_validator.js

```diff
@@ -3,7 +3,7 @@ import HttpError from "../http_error";
 import { isValidMongooseObjectId } from "../utils/validators";
 
 export default (req, res, next) => {
-  if (req.params.id && !isValidMongooseObjectId(req.params.id)) {
+  if (req.params.id === undefined || !isValidMongooseObjectId(req.params.id)) {
```

Comment:
requires an id for all requests it processes as middleware, cannot be  set as middleware on all requests